packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/impact-analysis/ImpactAnalysisDialogContext.tsx

```diff
@@ -30,43 +30,37 @@ export const ImpactAnalysisDialogContextProvider = ({ children }) => {
   const [state, setState] = useState<ImpactAnalysisDialogContextProviderState>({
     open: false,
     onConfirm: () => {},
-    editingContextId: null,
-    representationId: null,
-    toolId: null,
-    toolLabel: null,
-    diagramElementId: null,
-    variables: [],
+    toolLabel: '',
+    loading: false,
+    impactAnalysisReport: null,
   });
 
-  const showImpactAnalysisDialog = (
-    editingContextId: string,
-    representationId: string,
-    toolId: string,
+  const showImpactAnalysisDialog = <TVariables extends GQLInvokeImpactAnalysisVariables>(
+    query: LazyQueryResultTuple<GQLGetImpactAnalysisReportData, TVariables>,
+    queryParameters: TVariables,
     toolLabel: string,
-    diagramElementId: string,
-    variables: GQLToolVariable[],
     onConfirm: () => void
   ) => {
+    const [getImpactAnalysisReport, { loading, data }] = query;
+    getImpactAnalysisReport({
+      variables: queryParameters,
+    }).then((data) => {
+      setState((prevState) => ({
+        ...prevState,
+        loading: false,
+        impactAnalysisReport:
+          data.data?.viewer.editingContext.representation.description.diagramImpactAnalysisReport ?? null,
+      }));
+    });
     setState({
       open: true,
+      toolLabel: toolLabel,
+      loading,
+      impactAnalysisReport: data?.viewer.editingContext.representation.description.diagramImpactAnalysisReport ?? null,
       onConfirm,
-      editingContextId,
-      representationId,
-      toolId,
-      toolLabel,
-      diagramElementId,
-      variables,
     });
```

Comment:
This is the part I am skeptical about. 
The only way I have found to update the state when the query completes is to use `then`, which seems odd to me. I can't use `onCompleted`: it doesn't trigger, most likely because the palette is closed at that point (I tried to force the palette open and the `onCompleted` triggers).

I have the feeling there is a lifecycle issue here, and the way the query is created/passed to the context is not right, but I can't find the actual issue/how to solve it. Let me know if you have any lead.